changes/2023-09-26_encryption-context-decrypt/background.md

```diff
@@ -17,568 +17,139 @@ in this document are to be interpreted as described in
-
-var decryptOutput := esdk.Decrypt(Types.DecryptInput(
-   ciphertext := esdkCiphertext,
-   materialsManager := None,
-   keyring := Some(aesKeyring)
-   encryptionContext := Some(droppedRequiredKeyEncryptionContext)
-));
-
-expect decryptOutput.Failure?;
-```
-
-### Test Encryption Context has an extra pair on Decrypt Failure
-
-Encrypt will not store all encryption context.
-Decrypt will supply the key-value that was dropped but will also include a new key-value that
-was not used.
-This operation MUST fail.
-
-```
-// The string "asdf" as bytes
-var asdf := [ 97, 115, 100, 102 ];
-var aesKeyring := GetAesKeyring();
-
-// Test supply same encryption context on encrypt and decrypt NO filtering
-var encryptionContext := map[ keyA := valA, keyB := valB ];
-var requiredEncryptionContextKeys := [keyA];
-var reproducedEncryptionContext := map[ keyA := valA, keyC := valC ];
-
-var defaultCMM :- expect mpl.CreateDefaultCryptographicMaterialsManager(
-   mplTypes.CreateDefaultCryptographicMaterialsManagerInput(
-         keyring := aesKeyring
-   )
-);
-
-// Create Required EC CMM with the required EC Keys we want
-var reqCMM :- expect mpl.CreateRequiredEncryptionContextCMM(
-   mplTypes.CreateRequiredEncryptionContextCMMInput(
-         underlyingCMM := Some(defaultCMM),
-         keyring := None,
-         requiredEncryptionContextKeys := requiredEncryptionContextKeys
-   )
-);
-
-var encryptOutput := esdk.Encrypt(Types.EncryptInput(
-   plaintext := asdf,
-   encryptionContext := Some(encryptionContext),
-   materialsManager := Some(reqCMM),
-   keyring := None,
-   algorithmSuiteId := None,
-   frameLength := None
-));
-
-expect encryptOutput.Success?;
-var esdkCiphertext := encryptOutput.value.ciphertext;
-
-var decryptOutput := esdk.Decrypt(Types.DecryptInput(
-   ciphertext := esdkCiphertext,
-   materialsManager := None,
-   keyring := Some(aesKeyring)
-   encryptionContext := Some(reproducedEncryptionContext)
-));
-
-expect decryptOutput.Failure?;
-```
+To test the new encryption context (EC) features
+the following dimensions exists:
+
+- Stored EC -- EC stored in the header
+- Not stored EC -- EC authenticated to the header but not stored
+- CMM Material EC -- EC on the decryption material
+- Required keys -- Keys for the EC that MAY be only authenticated
+- Reproduced EC -- EC passed on decrypt
+
+Given the EC `{ a: a, b: b }`
+we break this into the following interesting options:
+
+Stored EC/Not Stored EC
+
+- `{a: a, b: b}` / `{}`
+- `{a: a}` / `{b: b}`
+- `{}` / `{a: a, b: b}`
+
+CMM Material/Required Keys
+
+- `{a: a, b: b}` / `{}`
+- `{a: a, b: b}` / `{a}`
+- `{a: a, b: b}` / `{a,b}`
+- `{a: a, b: c}` / `{a}`
+- `{a: a, b: b}` / `{c}`
+
+Reproduced EC
+
+- `{}`
+- `{ a: a }`
+- `{ b: b }`
+- `{ a: a, b: b }`
+- `{ a: c }`
+- `{ b: c }`
+- `{ a: c, b: b }`
+- `{ a: c, b: c }`
+- `{ c: c }`
+- `{ a: a, c: c }`
+- `{ b: b, c: c}`
+- `{ a: a, b: b, c: c }`
+
+### Message: `{a: a, b: b}` / `{}`
```

Comment:
I think we can further wordsmith these tables to simplify.
But this is a good start!